building_sim_plugins/building_sim_ign/src/door.cpp

```diff
@@ -180,91 +72,38 @@ class IGNITION_GAZEBO_VISIBLE DoorPlugin
     ignwarn << "Initializing plugin with name " << plugin_name << std::endl;
     _ros_node = std::make_shared<rclcpp::Node>(plugin_name);
 
-    RCLCPP_INFO(
-      _ros_node->get_logger(),
+    RCLCPP_INFO(_ros_node->get_logger(),
       "Loading DoorPlugin for [%s]",
       _model.Name(ecm).c_str());
+
     auto sdfClone = sdf->Clone();
 
-    MotionParams params;
-    get_sdf_param_if_available<double>(sdfClone, "v_max_door", params.v_max);
-    get_sdf_param_if_available<double>(sdfClone, "a_max_door", params.a_max);
-    get_sdf_param_if_available<double>(sdfClone, "a_nom_door", params.a_nom);
-    get_sdf_param_if_available<double>(sdfClone, "dx_min_door", params.dx_min);
-    // TODO add to traffic_editor
-    //get_sdf_param_if_available<double>(sdf, "f_max_door", params.f_max);
+    _door_common = DoorCommon::make(
+      _model.Name(ecm),
+      _ros_node,
+      sdfClone);
 
-    sdf::ElementPtr door_element;
-    std::string left_door_joint_name;
-    std::string right_door_joint_name;
-    std::string door_type;
-    if (!get_element_required(sdfClone, "door", door_element) ||
-      !get_sdf_attribute_required<std::string>(
-        door_element, "left_joint_name", left_door_joint_name) ||
-      !get_sdf_attribute_required<std::string>(
-        door_element, "right_joint_name", right_door_joint_name) ||
-      !get_sdf_attribute_required<std::string>(
-        door_element, "type", door_type))
-    {
-      RCLCPP_ERROR(
-        _ros_node->get_logger(),
-        " -- Missing required parameters for [%s] plugin",
-        _model.Name(ecm).c_str());
+    if (!_door_common)
       return;
-    }
-
-    if (left_door_joint_name != "empty_joint")
-    {
-      const auto left_door_joint =
-        _model.JointByName(ecm, left_door_joint_name);
-      if (!left_door_joint)
-      {
-        RCLCPP_ERROR(
-          _ros_node->get_logger(),
-          " -- Model is missing the left door joint [%s]",
-          left_door_joint_name.c_str());
-        return;
-      }
-      _doors.emplace_back(_model.Name(ecm) == "chart_lift_door",
-        left_door_joint, ecm, params);
-    }
 
-    if (right_door_joint_name != "empty_joint")
+    for (const auto& joint_name : _door_common->joint_names())
     {
-      const auto right_door_joint = _model.JointByName(ecm,
-          right_door_joint_name);
-      if (!right_door_joint)
+      const auto joint = std::make_shared<Entity>(
```

Comment:
Doesn't need to be a pointer

Reply:
It doesn't have to be a pointer, but I'm using one so that it is analogous to the JointPtr in `door.cpp` in building_gazebo_plugin. I can change it to `Entity` if you prefer.

Reply:
Not a pointer anymore.